node-js-basics-101/node_modules/asynquence-contrib/contrib.src.js

```diff
@@ -0,0 +1,2360 @@
+		for (i = 0; i < open.length; i++) {
+			(function iteration(action, channel, value) {
+				// put action?
+				if (Array.isArray(action)) {
+					channel = action[0];
+					value = action[1];
+
+					// define put handler
+					handlers.push(ASQ.iterable().then(function $$then() {
+						resolved = true;
+
+						// mark all handlers across this `alts(..)` as resolved now
+						handlers = handlers.filter(function $$filter(handler) {
+							return !(handler.resolved = true);
+						});
+
+						// channel still open?
+						if (!channel.closed) {
+							channel.messages.push(value);
+							isq.next({ value: true, channel: channel });
+						}
+						// channel already closed?
+						else {
+								isq.next({ value: false, channel: channel });
+							}
+					}));
+
+					// queue up put handler
+					channel.put_queue.push(handlers[handlers.length - 1]);
+
+					// take waiting on this queued put?
+					if (channel.take_queue.length > 0) {
+						schedule(function handleUnblocking() {
+							if (!resolved) {
+								unblock(channel.put_queue.shift());
+								unblock(channel.take_queue.shift());
+							}
+						}, 0);
+					}
+				}
+				// take action?
+				else {
+						channel = action;
+
+						// define take handler
+						handlers.push(ASQ.iterable().then(function $$then() {
+							resolved = true;
+
+							// mark all handlers across this `alts(..)` as resolved now
+							handlers = handlers.filter(function $$filter(handler) {
+								return !(handler.resolved = true);
+							});
+
+							// channel still open?
+							if (!channel.closed) {
+								isq.next({ value: channel.messages.shift(), channel: channel });
+							}
+							// channel already closed?
+							else {
+									isq.next({ value: ASQ.csp.CLOSED, channel: channel });
+								}
+						}));
+
+						// queue up take handler
+						channel.take_queue.push(handlers[handlers.length - 1]);
+
+						// put waiting on this queued take?
+						if (channel.put_queue.length > 0) {
+							schedule(function handleUnblocking() {
+								if (!resolved) {
+									unblock(channel.put_queue.shift());
+									unblock(channel.take_queue.shift());
+								}
+							});
+						}
+					}
+			})(open[i]);
+		}
+
+		return ret;
+	}
+
+	function altsAsync(chans, cb) {
+		var ret = ASQ(alts(chans));
+
+		if (cb && typeof cb == "function") {
+			ret.pThen(cb, cb);
+		} else {
+			return ret;
+		}
+	}
+
+	function timeout(delay) {
+		var ch = channel();
+		setTimeout(ch.close, delay);
+		return ch;
+	}
+
+	function go(gen, args) {
+		// goroutine arguments passed?
+		if (arguments.length > 1) {
+			if (!args || !Array.isArray(args)) {
+				args = [args];
+			}
+		} else {
+			args = [];
+		}
+
+		return regeneratorRuntime.mark(function $$go(token) {
+			var unblock, ret, msg, err, type, done, it;
+			return regeneratorRuntime.wrap(function $$go$(context$3$0) {
+				while (1) switch (context$3$0.prev = context$3$0.next) {
+					case 0:
+						unblock = function unblock() {
```

Comment:
__[jshint]__
* Functions declared within loops referencing an outer scoped variable may lead to confusing semantics.

<!-- Comment by [SideCI](https://sideci.com) -->